lib/core/src/Cardano/Wallet.hs

```diff
@@ -1536,6 +1539,29 @@ listTransactions ctx wid mStart mEnd order = db & \DBLayer{..} -> do
         _ ->
             pure $ slotRangeFromTimeRange sp $ Range mStart mEnd
 
+-- | Get transaction and metadata from history for a given wallet.
+getTransaction
+    :: forall ctx s k. HasDBLayer s k ctx
+    => ctx
+    -> WalletId
+    -> Hash "Tx"
+    -> ExceptT ErrGetTransaction IO TransactionInfo
+getTransaction ctx wid tid = db & \DBLayer{..} -> do
+    let pk = PrimaryKey wid
+    mapExceptT atomically $ do
+        _ <- withExceptT ErrGetTransactionNoSuchWallet $
+            withNoSuchWallet wid $ readCheckpoint pk
+        txs <- lift $ readTxHistory pk Descending wholeRange Nothing
+        case (filter txPresent txs) of
+            [] -> do
```

Comment:
Arg! This defies a bit the purpose of such endpoint. Being able to fetch a single transaction should be quick and swift; here we are loading the entire transaction history in memory and filtering it by hand! For exchanges processing MANY transactions a day, that's not going to work (especially because they'll be the primary consumer of this endpoint). 

This should be replaced by a single SQL query to retrieve a single transaction from the database :+1:

Reply:
introduced getTx in  https://github.com/input-output-hk/cardano-wallet/pull/1795/commits/d898b0d84bba687166dcd71bfd440d29cd26f063 and https://github.com/input-output-hk/cardano-wallet/pull/1795/commits/9f5eba327dde35772a26d451d35c0718f3524341